tests/test_app/tests.py

```diff
@@ -10,3 +16,75 @@
 
 def test_dummy():
     assert 1 == 1
+
+
+def test_token_create_serializer_get_token():
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    serializer = TokenCreateSerializer(user)
+    token = serializer.get_token(user)
+    assert isinstance(token, AccessToken)
+    assert token['token_type'] == 'access'
+    assert token['user_id'] == user.id
+    assert 'exp' in token
+    assert 'jti' in token
+
+
+def test_token_create_serializer_validate():
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    payload = {'email':'test@test.com','password':'12345'}
+    serializer = TokenCreateSerializer(data=payload)
+    serializer.user=user
+    assert serializer.is_valid(raise_exception=True) is True
+
+def test_token_obtain_serializer_get_fields():
+    serializer = TokenObtainSerializer()
+    expected_fields = ['email', 'phone_number','password','otp','token']
+    fields = serializer.get_fields()
+    assert len(fields.keys()) == len(expected_fields)
+    assert all([field in fields.keys() for field in expected_fields])
+
+
+def test_token_obtain_serializer_validate():
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    payload = {'email':'test@test.com','password':'12345'}
+    serializer = TokenObtainSerializer(data=payload)
+    serializer.user=user
+    assert serializer.is_valid(raise_exception=True) is True
```

Comment:
Additional validation is needed here. Please note that the validation depends on:
- `api_settings.REQUIRED_AUTH_FIELDS`, the serializer should throw an error if these fields are not provided. 
- `api_settings.OPTIONAL_AUTH_FIELDS`, the serializer should be able to tolerate empty data for these fields.

So, we should patch the `api_settings.REQUIRED_AUTH_FIELDS` and see if that is affecting the serializer validation as expected. The same goes for `api_settings.OPTIONAL_AUTH_FIELDS` as well.